extensions/charts/vsphereoutoftree.go

```diff
@@ -0,0 +1,178 @@
+package charts
+
+import (
+	"github.com/rancher/shepherd/extensions/projects"
+	"github.com/rancher/shepherd/extensions/rke1/nodetemplates"
+	"github.com/rancher/shepherd/pkg/api/steve/catalog/types"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	r1vsphere "github.com/rancher/shepherd/extensions/rke1/nodetemplates/vsphere"
+)
+
+const (
+	systemProject       = "System"
+	vsphereCPIchartName = "rancher-vsphere-cpi"
+	vsphereCSIchartName = "rancher-vsphere-csi"
+
+	vcenter      = "vCenter"
+	storageclass = "storageClass"
+
+	datacenters  = "datacenters"
+	host         = "host"
+	password     = "password"
+	username     = "username"
+	port         = "port"
+	clusterid    = "clusterId"
+	datastoreurl = "datastoreURL"
+)
+
+// InstallVsphereOutOfTreeCharts installs the CPI and CSI chart for aws cloud provider in a given cluster.
```

Comment:
My other question here; installing these charts separated and one by one doesn't make sense right? Both need to be installed?

Reply:
correct, both need to be installed.